### PR TITLE
build(.github): fix picking the minor version branch

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR fixes picking the right minor version branch, and also disables the fail fast behavior of the matrix strategy, allowing specific QA jobs to keep going even if one fails.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
Test run: https://github.com/camunda/zeebe/actions/runs/3512268668

**Comment by npepinpe** [comment]
Final test run: https://github.com/camunda/zeebe/actions/runs/3512601359/

:crossed_fingers:

**Comment by npepinpe** [comment]
Latest run looks good :+1:

File: .github/workflows/daily-qa.yml [hunk 1]
```diff
@@ -50,6 +50,8 @@ jobs:
     needs:
       - get-versions
     strategy:
+      # do not cancel other jobs if one fails
+      fail-fast: false
```
Comment (oleschoenburg) [inline .github/workflows/daily-qa.yml:54]: :+1:

File: .github/workflows/daily-qa.yml [hunk 2]
```diff
@@ -32,15 +32,15 @@ jobs:
       - uses: actions/checkout@v3
       - run: git fetch --tags
       - run: |
-          latest=$(git tag | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' | sort -Vr | head -n1)
-          previous=$(git tag | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' | grep -Ev "^${latest%\.[0-9]}.*$" | sort -Vr | head -n1)
+          latest=$(git tag | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' | sed -E -e 's/\.[0-9]+$//' | sort -Vr | head -n1)
```
Comment (oleschoenburg) [inline .github/workflows/daily-qa.yml:35]: :wrench: Took me a moment to figure out what's going on here. Maybe a comment like "lists all version tags, drops the patch version and takes the latest. For previous version, do the same but exclude the latest version from candidates"

**Review by lenaschoenburg — APPROVED** [review]
Thanks, looks good to me :) :+1:

File: .github/workflows/daily-qa.yml (hunk 2 above)
Comment (npepinpe) [reply .github/workflows/daily-qa.yml:35]: I think it's not visible in the PR, but the comment above the job explains this. I can make it more explicit though :+1:

Comment (oleschoenburg) [reply .github/workflows/daily-qa.yml:35]: Oh yeah, that's perfectly fine :+1:

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**